freeipa/src/main/java/com/sequenceiq/freeipa/flow/instance/reboot/RebootService.java

```diff
@@ -0,0 +1,39 @@
+package com.sequenceiq.freeipa.flow.instance.reboot;
+
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.event.instance.RebootInstancesResult;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.instance.InstanceStatus;
+import com.sequenceiq.freeipa.service.stack.RebootInstancesService;
+import com.sequenceiq.freeipa.service.stack.instance.InstanceMetaDataService;
+
+@Component
+public class RebootService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RebootService.class);
+
+    @Inject
+    private InstanceMetaDataService instanceMetaDataService;
+
+    @Inject
+    private RebootInstancesService rebootInstancesService;
+
+    public void startInstanceReboot(RebootContext context) {
+        instanceMetaDataService.updateStatus(context.getStack(), context.getInstanceIdList(), InstanceStatus.REBOOTING);
+    }
+
+    public void handleInstanceRebootError(String instanceIds) {
```

Comment:
so we would leave the instances in rebooting state in case of an error?

Reply:
fixed.